dotnet/samples/Concepts/AgentSyntax/Getting_Started/Step4_KernelFunctionStrategies.cs

```diff
@@ -0,0 +1,130 @@
+    {
+        // Define the agents
+        ChatCompletionAgent agentReviewer =
+            new()
+            {
+                Instructions = ReviewerInstructions,
+                Name = ReviewerName,
+                Kernel = this.CreateKernelWithChatCompletion(),
+            };
+
+        ChatCompletionAgent agentWriter =
+            new()
+            {
+                Instructions = CopyWriterInstructions,
+                Name = CopyWriterName,
+                Kernel = this.CreateKernelWithChatCompletion(),
+            };
+
+        KernelFunction terminationFunction =
+            KernelFunctionFactory.CreateFromPrompt(
+                $$$"""
+                Determine if the copy has been approved.  If so, respond with a single word: yes
+
+                History:
+                {{${{{KernelFunctionTerminationStrategy.HistoryArgumentName}}}}}
+                """);
+
+        KernelFunction selectionFunction =
+            KernelFunctionFactory.CreateFromPrompt(
+                $$$"""
+                You are in a role playing game.
+                Carefully read the conversation history and carry on the conversation by specifying only the name of player to take the next turn.
+
+                The available names are:
+                {{${{{KernelFunctionSelectionStrategy.AgentsArgumentName}}}}}
+
+                History:
+                {{${{{KernelFunctionSelectionStrategy.HistoryArgumentName}}}}}
+                """);
+
+        // Create a chat for agent interaction.
+        AgentGroupChat chat =
+            new(agentWriter, agentReviewer)
+            {
+                ExecutionSettings =
+                    new()
+                    {
+                        // Here KernelFunctionTerminationStrategy will terminate
+                        // when the art-director has given their approval.
+                        TerminationStrategy =
+                            new KernelFunctionTerminationStrategy(terminationFunction)
+                            {
+                                // Only the art-director may approve.
+                                Agents = [agentReviewer],
+                                // Kernel utilized when invoking the kernel-function.
+                                Kernel = CreateKernelWithChatCompletion(),
+                                // Customer result parser to determine if the response is "yes"
+                                ResultParser = new AffirmativeFunctionResultParser(),
+                            },
+                        // Here a KernelFunctionSelectionStrategy selects agents based
+                        // on a prompt function.
+                        SelectionStrategy =
+                            new KernelFunctionSelectionStrategy(selectionFunction)
+                            {
+                                // Kernel utilized when invoking the kernel-function.
+                                Kernel = CreateKernelWithChatCompletion()
```

Comment:
Do we need seperate instances of `Kernel`?

Reply:
Not necessarily.  It was my understanding that `Kernel` is intended to be lightweight and not treated as a singleton.  All of  the examples follow this idea.  Using the same kernel for all articulation points means the same filters will be called by call components, same plugins, etc...  While this consideration isn't necessarily in play for this example, I do wonder if it isn't somewhat misleading to initialize all the agent components with the same `Kernel` in a reference example.

I suppose the reverse question may be: it is better to demo with a shared instance?

Pro: Trivially one less `Kernel` instance
Con: Obscures the value in targeting various agent components with different kernels

What do you think?

Reply:
OK, yeah that makes sense. Whatever pattern we use the samples will be copied so showing most flexible approach makes sense.